File: src/SIL.XForge.Scripture/Models/BiblicalTerm.cs
using System.Collections.Generic;
using SIL.XForge.Models;

namespace SIL.XForge.Scripture.Models;

public class BiblicalTerm : ProjectData
{
    public string DataId { get; set; } = string.Empty;
    public string TermId { get; set; } = string.Empty;
    public string Transliteration { get; set; } = string.Empty;
    public IList<string> Renderings { get; set; } = [];
    public string Description { get; set; } = string.Empty;
    public string Language { get; set; } = string.Empty;
    public IList<string> Links { get; set; } = [];
    public IList<int> References { get; set; } = [];
    public Dictionary<string, BiblicalTermDefinition> Definitions { get; set; } = [];
}

File: src/RealtimeServer/scriptureforge/models/biblical-term.ts
import { PROJECT_DATA_INDEX_PATHS, ProjectData } from '../../common/models/project-data';
import { BiblicalTermDefinition } from './biblical-term-definition';

export const BIBLICAL_TERM_COLLECTION = 'biblical_terms';
export const BIBLICAL_TERM_INDEX_PATHS = PROJECT_DATA_INDEX_PATHS;

export function getBiblicalTermDocId(projectId: string, dataId: string): string {
  return `${projectId}:${dataId}`;
}

export interface BiblicalTerm extends ProjectData {
  dataId: string;
  termId: string;
  transliteration: string;
  renderings: string[];
  description: string;
  language: string;
  links: string[];
  references: number[];
  definitions: { [language: string]: BiblicalTermDefinition };
}
